platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/internal/cc/impl/isolated/IsolatedProjectsCompositeBuildIntegrationTest.groovy

```diff
@@ -47,4 +51,122 @@ class IsolatedProjectsCompositeBuildIntegrationTest extends AbstractIsolatedProj
         then:
         fixture.assertStateLoaded()
     }
+
+    def "cycles for plugin builds are prohibited"() {
+        given:
+        includePluginBuild(settingsFile, "plugins-a")
+        includedBuild("plugins-a") { layout ->
+            includePluginBuild(layout.settingsScript, "../plugins-b")
+        }
+        includedBuild("plugins-b") { layout ->
+            includePluginBuild(layout.settingsScript, "../plugins-c")
+        }
+        includedBuild("plugins-c") { layout ->
+            includePluginBuild(layout.settingsScript, "../plugins-a")
+        }
+
+        when:
+        isolatedProjectsFails("help")
+
+        then:
+        failureDescriptionContains("A cycle has been detected in the definition of plugin builds: :plugins-c -> :plugins-a -> :plugins-b -> :plugins-c. This is not supported with Isolated Projects. Please update your build definition to remove one of the edges.")
+    }
+
+    def "transitive cycles for plugin builds are prohibited"() {
```

Comment:
🧪 It seems we're missing a test for `library-build-a -> plugin-build -> library-build-b -> library-build-a`, i.e., a cycle starting and ending in a library build which goes through a plugin build.

Reply:
@bamboo For catch this we have to prohibit library cycles as well. We anyway have to do that tho.

Reply:
UPD. Not really, actually. The trick is report cycles not only when we include plugin build, but also when plugin build is including other build